include/picongpu/fields/incidentField/profiles/DispersivePulse.hpp

```diff
@@ -299,15 +299,39 @@ namespace picongpu
                             float_X const dt = static_cast<float_X>(picongpu::SI::DELTA_T_SI / UNIT_TIME);
                             // interpolation order
                             float_X N_raw = Unitless::INIT_TIME / dt;
-                            int n = static_cast<int>(N_raw * 0.5_X); // -0 instead of -1 for rounding up N_raw
+                            int const n = static_cast<int>(N_raw * 0.5_X); // -0 instead of -1 for rounding up N_raw
+
+                            // frequency step for DFT
+                            float_X const dOmk = pmacc::math::Pi<float_X>::doubleValue / Unitless::INIT_TIME;
+
+                            // Since the (Gaussian) spectrum has only significant values near the central frequency,
+                            // the summation over all frequencies is reduced to frequencies within an interval
+                            // around the central frequency.
+
+                            // standard deviation of the Gaussian distributed spectrum
+                            float_X const sigma_Om = 1._X / (pmacc::math::sqrt(2._X) * Unitless::PULSE_DURATION);
+
+                            // index of the mean frequency of the Gaussian distributed spectrum
+                            // unit: [dOmk]
+                            int const center_k
+                                = static_cast<int>(SPEED_OF_LIGHT * Unitless::INIT_TIME / Unitless::Base::WAVE_LENGTH);
+
+                            // index of the lowest frequency in the Gaussian distributed spectrum which is used in the
+                            // DFT 4*sigma_Om distance from central frequency
+                            int const minOm_k = center_k - static_cast<int>(4._X * sigma_Om / dOmk);
+                            int const k_min = minOm_k < 1 ? 1 : minOm_k;
+
+                            // index of the highest frequency in the Gaussian distributed spectrum which is used in the
+                            // DFT
+                            int const maxOm_k = 2 * center_k - minOm_k;
+                            int const k_max = maxOm_k <= n ? maxOm_k : n;
 
                             float_X E_t = 0.0_X; // electric field in time-domain
-                            float_X Omk = 0.0_X; // stores angular frequency for DFT-loop; Omk= 2pi*k/T
+                            float_X Omk = 0.0_X; // stores angular frequency for DFT-loop; Omk = 2pi*k/T = dOmk * k
 
-                            for(int k = 1; k < n + 1; k++)
+                            for(int k = k_min; k < k_max + 1; k++)
```

Comment:
Did you check the output especially for high AD and/or TOD values? Since they scale with $(\Omega - \Omega_0)^3$, there is a high contribution away from `center_k` (definitely less than the exponential decay of the envelope, nevetheless I just want to be sure)

Reply:
could it be that it must be `<= k_max`?

Reply:
ohh I see you have `k < k_max + 1` what is equal to `k <= k_max`. I prefere the second version.

Reply:
@fabidie AD or TOD does not change the spectral width. Spectral width is only determined by the pulse duration and this is used to calculate the spectral width. Therefore all should be fine the way it is?